### PR TITLE
Added basic admin API service to the gateway.

### PR DESCRIPTION
**Pull request by prototypo**

### CODE REVIEW

File: internal/api/adminapi/admin_api.go [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+			// Error in tcp communication, drop the connection.
+			logging.Error1(err)
+			return
+		}
+	}
+}
+
+// GetConnForRequestingAdminClient returns the connection for sending request to an admin client with given id.
+// It will reuse any active connection.
+func GetConnForRequestingAdminClient(gatewayID nodeid.NodeID, g *gateway.Gateway) (*gateway.CommunicationChannel, error) {
```
Comment (wcgcyx) [inline internal/api/adminapi/admin_api.go:92]: We probably won't need this for admin API.

**Review by wcgcyx — CHANGES_REQUESTED** [review]
Others look good to me.

File: internal/api/adminapi/get_reputation_test.go [hunk 2]
```diff
@@ -0,0 +1,127 @@
+package reputation
```
Comment (drinkcoffee) [inline internal/api/adminapi/get_reputation_test.go:1]: The package for this should be adminapi

File: internal/api/adminapi/get_reputation_test.go [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+
```
Comment (drinkcoffee) [inline internal/api/adminapi/get_reputation_test.go:2]: The test code appears to be identical to the test code in the reputation package. Should this file have been committed?

**Review by drinkcoffee — CHANGES_REQUESTED** [review]

File: internal/api/adminapi/get_reputation_test.go (hunk 3 above)
Comment (prototypo) [reply internal/api/adminapi/get_reputation_test.go:2]: This file never should have been committed - it was my mistake. I will remove it.

File: internal/api/adminapi/admin_api.go (hunk 1 above)
Comment (prototypo) [reply internal/api/adminapi/admin_api.go:92]: Peter suggested keeping it for the future, so I guess we will.

File: internal/api/adminapi/get_reputation_test.go (hunk 3 above)
Comment (prototypo) [reply internal/api/adminapi/get_reputation_test.go]: The file has been deleted.

File: internal/api/adminapi/get_reputation_test.go (hunk 2 above)
Comment (prototypo) [reply internal/api/adminapi/get_reputation_test.go]: Deleted because this it never should have been committed - it was my mistake.

**Pull request merged**